kak-lsp.toml

```diff
@@ -351,47 +326,76 @@
         fi
     """,
 ]
-settings_section = "rust-analyzer"
-[language.rust.settings.rust-analyzer]
+[language_server.rust-analyzer.settings.rust-analyzer]
 # See https://rust-analyzer.github.io/manual.html#configuration
 hoverActions.enable = false # kak-lsp doesn't support this at the moment
 # cargo.features = []
 
-[language.terraform]
+[language_server.solargraph]
+filetypes = ["ruby"]
+roots = ["Gemfile"]
+command = "solargraph"
+args = ["stdio"]
+settings_section = "_"
+[language_server.solargraph.settings._]
+# See https://github.com/castwide/solargraph/blob/master/lib/solargraph/language_server/host.rb
+# diagnostics = false
+
+[language_server.terraform-ls]
 filetypes = ["terraform"]
 roots = ["*.tf"]
 command = "terraform-ls"
 args = ["serve"]
-[language.terraform.settings.terraform-ls]
+[language_server.terraform-ls.settings.terraform-ls]
 # See https://github.com/hashicorp/terraform-ls/blob/main/docs/SETTINGS.md
 # rootModulePaths = []
 
-[language.tsx] # works for typescript as well
-filetypes = ["typescript"]
-roots = ["package.json", "tsconfig.json", ".git", ".hg"]
+[language_server.texlab]
+filetypes = ["latex"]
+roots = [".git", ".hg"]
+command = "texlab"
+[language_server.texlab.settings.texlab]
```

Comment:
I wonder if we should keep sorting by "primary filetype".
However I'm not sure, so no need to change this now.